extensions/wikibase/src/org/openrefine/wikibase/qa/EditInspector.java

```diff
@@ -226,26 +233,50 @@ private void resolveWarningPropertyLabels() throws ExecutionException {
                                 if (propertyDocument != null && propertyDocument.getLabels() != null) {
                                     label = propertyDocument.getLabels().get(Locale.getDefault().getLanguage()) != null
                                             ? propertyDocument.getLabels().get(Locale.getDefault().getLanguage()).getText()
                                             : "";
                                 }
-
-                                // Update the property with a new SuggestedPropertyIdValue
                                 warning.setProperty(key, new SuggestedPropertyIdValue(property.getId(), property.getSiteIri(), label));
                             }
                         });
                     });
+
+            warningStore.getWarnings().stream().forEach(warning -> {
+                if (warningTypeswithSuggestedData.contains(warning.getType())) {
+                    PropertyIdValue parentProperty = (PropertyIdValue) warning.getProperties().get("property_entity");
+                    PropertyIdValue relatedProperty = (PropertyIdValue) warning.getProperties().get("added_property_entity");
+                    PropertyDocument propertyDocument = (PropertyDocument) entityCache.get(parentProperty);
+                    if (propertyDocument != null) {
+                        propertyDocument.getAllStatements().forEachRemaining(statement -> {
+                            if (statement.getClaim().getMainSnak().getPropertyId().getId().equals(relatedProperty.getId())) {
+                                ItemDocument itemDocument = (ItemDocument) entityCache.get(((EntityIdValue) statement.getValue()));
+
+                                String itemLabel = "";
+                                if (itemDocument != null && itemDocument.getLabels() != null) {
+                                    itemLabel = itemDocument.getLabels().get(Locale.getDefault().getLanguage()) != null
+                                            ? itemDocument.getLabels().get(Locale.getDefault().getLanguage()).getText()
+                                            : "";
+                                }
+                                SuggestedPropertyIdValue current = (SuggestedPropertyIdValue) warning.getProperties()
+                                        .get("added_property_entity");
+                                SuggestedItemIdValue itemIdValue = new SuggestedItemIdValue(((EntityIdValue) statement.getValue()).getId(),
+                                        current.getSiteIri(), itemLabel);
+                                SuggestedPropertyIdValue updated = new SuggestedPropertyIdValue(current.getId(), current.getSiteIri(),
+                                        current.getLabel(), itemIdValue);
+                                warning.setProperty("added_property_entity", updated);
+                            }
+                        });
+                    }
+                }
+            });
```

Comment:
I wouldn't have expected this new block of code to be necessary: I would just have added support for `ItemIdValue` alongside `PropertyIdValue` above.

Reply:
Removed block